lib/sgtn-client/core/config.rb

```diff
@@ -165,7 +165,27 @@ def loader
             SgtnClient::TranslationLoader::LoaderFactory.create(config)
           end
         end
-      
+
+        def available_bundles
+          loader.available_bundles
+        rescue StandardError => e
+          SgtnClient.logger.error 'failed to get available bundles'
+          SgtnClient.logger.error e
+          {}
+        end
+
+        def available_locales
+          bundles = loader.available_bundles
+          return {} unless bundles
+
+          unless bundles.respond_to?(:locales)
+            def bundles.locales
+              @locales ||= reduce(Set.new) { |locales, id| locales << id.locale }
+            end
+          end
+          bundles.locales
+        end
```

Comment:
In what case would bundles.respond_to(:locales) be true?
```suggestion
            def bundles.locales
              @locales ||= reduce(Set.new) { |locales, id| locales << id.locale }
            end
          end
          bundles.locales
```

Reply:
After cache is expired, available bundles will be updated. Then `bundles` don't respond to `:locales` because it's new.
Need to run the code lines from 182 to 184 to add a method `locales` for `bundles`.

Reply:
I tested and it always runs lines 182-184 even when bundles is not expired yet.

Reply:
> I tested and it always runs lines 182-184 even when bundles is not expired yet.

Yes. You are right! My last reply has some problems. 

The block 181~185 is executed only once before cache is expired. It is to define the `locales` method.
In the following calls, will execute 182~184 each time. It is the `locales` method body.
The `locales` method won't be defined twice.

Reply:
Lines 181-185 executes for every getString call. I tested this by simply printing something before line 183. This is because you are always calling bundles.locales in line 186. See the new code suggestion to avoid executing the code multiple times.

Reply:
This is to utilize cache. After cache is expired, locales can be updated too.
If save locales to an instance var of Config, locale list can't be update after cache is expired.
How do you think? I'm fine with both.

Reply:
I understand now. I missed the `||=` on `@locales` in line 183. 
182-184 executes on every getString call but the `reduce` method does not because `@locales` is already populated. This works.